### PR TITLE
REF: AT1K4 transmission calculations on PLC [LCLSPC-208]

### PR DESCRIPTION
**Pull request by klauer**
We have determined it's not acceptable for PMPS to have a Python process calculating transmission values as [solid-attenuator](https://github.com/pcdshub/solid-attenuator) is currently doing in the case of AT2K2. 

As such, using a new dependency [lcls-twincat-physics](https://github.com/pcdshub/lcls-twincat-physics), the solid attenuators must calculate transmission on the PLC to coordinate with PMPS at PLC rates.

I found that these calculations were working well enough for AT2K2, so the next logical step was AT1K4. AT2L0 will require some custom handling after this, as it is a different beast.

A few things:
* So far, this builds but I haven't updated the tmc or IOC
* I'd like some help in triple-checking that I have the filter information correct. I'll have to remind reviewers about this as it's rather important.

### CODE REVIEW

File: plc-kfe-motion/kfe_motion/POUs/AT1K4/FB_AttenuatorElementDensity.TcPOU [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+END_VAR
+VAR
+    fbElementDensity : FB_ElementDensity;
+    fDensity_gm3 : LREAL;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[IF sName = 'C' THEN
+    (* Special-case diamond here *)
+    fDensity := 3.51E6;  (* C (Diamond) g/m^3 *)
```
Comment (ZLLentz) [inline plc-kfe-motion/kfe_motion/POUs/AT1K4/FB_AttenuatorElementDensity.TcPOU:20]: Any story behind the special case for diamond?

File: plc-kfe-motion/kfe_motion/POUs/AT1K4/FB_SXR_SATT_Stage.TcPOU [hunk 2]
```diff
@@ -18,12 +18,43 @@ VAR_INPUT
...
 END_VAR
 VAR_OUTPUT
     // stExtra : DUT_MotionStage_Extras;
     fTemp1 : LREAL;
     fTemp2 : LREAL;
     bIsStationary : BOOL;
+    bError : BOOL;
+
+    {attribute 'pytmc' := '
+        pv: Material
```
Comment (ZLLentz) [inline plc-kfe-motion/kfe_motion/POUs/AT1K4/FB_SXR_SATT_Stage.TcPOU:34]: Duplicate comment from other review: should these be e.g. `MATERIAL`?

**Review by ZLLentz — APPROVED** [review]
Confirmed that the filter thicknesses, materials, positions are consistent with https://confluence.slac.stanford.edu/display/L2SI/AT1K4-SOLID+SXR+Solid+Attenuator

File: plc-kfe-motion/kfe_motion/POUs/AT1K4/FB_AttenuatorElementDensity.TcPOU (hunk 1 above)
Comment (klauer) [reply plc-kfe-motion/kfe_motion/POUs/AT1K4/FB_AttenuatorElementDensity.TcPOU:20]: Density of carbon depends on the structure; and periodictable which is behind lcls-twincat-physics I think assumes amorphous:

image

**Pull request merged**